swiftshield-Sources/AutomaticSwiftShield.swift

```diff
@@ -118,26 +135,43 @@ extension AutomaticSwiftShield {
             return nil
         }
         
-        if self.filesToIgnore.contains(name) {
-            return nil
-        }
-        
-        if self.excludedPrefixTag != "" && name.hasPrefix(self.excludedPrefixTag) {
-            return nil
-        }
-        
-        if self.excludedSuffixTag != "" && name.hasSuffix(excludedSuffixTag) {
-            return nil
+        if excludePublic {
+            let attributesDict = SKApi.sourcekitd_variant_dictionary_get_value(dict, sourceKit.attributesID)
+            let attributesData = dict.getAttributes(dict: attributesDict, subKey: sourceKit.attributeID)
+
+            
+            //Check if variant is public
+            let isPublic = attributesData.filter { item in
+                return item.contains("public")
+            }.count != 0
+            
+            //Add to publicProtocols array
+            if kind == "source.lang.swift.decl.protocol" && isPublic {
+                publicProtocols.insert(name)
+            }
+            
+            //Don't Obfuscate public methods/properties..
+            if isPublic {
+                return nil
+            }
+            
+            //Handle public protocol's functions
+            for protocolName in publicProtocols {
+                if usr.contains(protocolName) {
```

Comment:
If I understood this correctly, shouldn't this be simply `publicProtocols.contains(usr)` instead of a loop? This is running in O(nˆ2)

Reply:
No, because the `receiver_usr` isn't returning just a name, It's something like that 
`s:17ObfuscationSource14ViewControllerC5countSivp`

Reply:
I see now. The problem is that using `contains` inside an usr will likely not work in a bigger app because nothing stops other modules from having private protocols with the same name as your public one. I think you need to store the protocol usrs in `publicProtocols` instead of their names to avoid this loop.

But I'm getting some bad vibes, is this deterministic? I'm having the impression that a public protocol's method would be wrongly obfuscated if the order of the files changed. If you look at `isReferencingInternal()`, I had to even make it recursive to make it work. Things are slightly more difficult here because it happens before everything, but I think it's gonna work nicely in the end.